src/snovault/elasticsearch/searches/fields.py

```diff
@@ -662,3 +663,29 @@ def render(self, *args, **kwargs):
         return {
             SEARCH_BASE: self._get_search_base()
         }
+
+
+class FacetGroupsResponseField(ResponseField):
+    '''
+    Passes through the `facet_groups` property from the schema to the search
+    results, adding the relevant schema name as the `name` property for each
+    facet group.
+    '''
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def _get_facet_groups(self):
+        facet_groups = []
+        for config in self.get_query_builder()._get_configs_from_param_values_or_item_types():
+            schema_facet_groups = config.get('facet_groups', [])
+            for schema_facet_group in schema_facet_groups:
+                schema_facet_group['name'] = config.name
+            facet_groups.extend(schema_facet_groups)
```

Comment:
Concerned this is actually modifying registry. Probably safer just to append a newly built up dictionary to the list. Something like:
```
for schema_facet_group in schema_facet_groups:
    facet_groups.append(
        {
            'name': config.name,
            **schema_facet_group,
        }
    )
```
should avoid mutating anything in place and avoids another loop to extend. (The `**` syntax destructures the dict like `...` in JS).